### PR TITLE
BackupBrowser: Add file details scaffolding and row counts for database tables

### PR DESCRIPTION
**Pull request by Initsogar**
We are currently working on the API to fetch files info when clicking on them while navigating on the backup browser. In the meantime, we are including the number of rows when clicking a database table (`/sql/*`) as we don't require an additional API to get that information.

## Proposed Changes

* Add `FileInfoCard` component, which will be in charge of rendering the file details depending on the item type. For now, it only works for database tables.
* Add the number of rows backed up when a user clicks on a database table.
* Refactor the way we handle clicks on items. Now all items could be opened, not just the ones with `hasChildren === true`, but only those will render children nodes.
* Add a `activeNodePath` state in `FileBrowser` so we could track the current node opened. This will help us to expand the file details on one node when clicked and collapse the previous opened.

## Screenshot

image

## Testing Instructions

* Start a Jetpack Cloud or Calypso Live branch
* Select one of your sites with a Jetpack VaultPress Backup plan.
* Navigate to VaultPress Backup page
* Pick one date with a backup available (use the calendar and click on any date marked with a dot).
* Browse the `/sql/` directory and clicks on any database table. It should open a card with the number of rows that table includes.
* Browse other directories and ensure clicking on an item doesn't display the info card.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/backup/backup-contents-page/file-browser/file-browser-node.tsx [hunk 1]
```diff
@@ -23,7 +24,6 @@ const FileBrowserNode: FunctionComponent< FileBrowserNodeProps > = ( {
 	isAlternate,
 } ) => {
 	const isRoot = path === '/';
-	const hasChildren = item.hasChildren;
```
Comment (Initsogar) [inline client/my-sites/backup/backup-contents-page/file-browser/file-browser-node.tsx:26]: This `hasChildren` constant was redundant. We could use `item.hasChildren` directly in the two instances we need it in this component.

File: client/my-sites/backup/backup-contents-page/file-browser/file-info-card.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+	return (
+		<div className="file-card">
+			<div className="file-card__details">
+				<div className="file-card__row">
+					<span className="file-card__label">Rows: </span>
+					<span className="file-card__value">{ item.rowCount }</span>
+				</div>
+			</div>
+			<div className="file-card__actions"></div>
```
Comment (Initsogar) [inline client/my-sites/backup/backup-contents-page/file-browser/file-info-card.tsx:30]: Here will be the `download` button :D

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8061391

Thank you @Initsogar for including a screenshot in the description! This is really helpful for our translators.

**Comment by jwebbdev** [comment]
We might want to check with design on whether we should have more than one info open at once.  VaultPress legacy only has one open at a time and closes other info windows as you open a new one.

Screenshot 2023-06-22 at 10 54 51 AM

**Comment by Initsogar** [comment]
> We might want to check with design on whether we should have more than one info open at once. VaultPress legacy only has one open at a time and closes other info windows as you open a new one.

Nice catch! I would say it is better to have only one open as a legacy, especially if we are going to include file previews as well. I will work on it.

**Comment by Initsogar** [comment]
Just pushed changes addressing the issue of having more than one file details open.

**Review by jwebbdev — APPROVED** [review]
Looks good.

Elements without info will now close elements with info if clicked, but since they'll all have info soon I think it's probably not worth solving just to undo.

**Comment by Initsogar** [comment]
> Elements without info will now close elements with info if clicked, but since they'll all have info soon I think it's probably not worth solving just to undo.

Correct, forgot to mention it as a "known issue", but it is actually how it should behave when we have file details for all item and will be fixed in an upcoming PR.

Thanks for the review! Merging :)

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8061391).

**Pull request merged**